### PR TITLE
fix: index message_id column of transactional SMS

### PR DESCRIPTION
**Pull request by tan-zx**
[Issue](https://opengovproducts.slack.com/archives/C015VQQ3HMG/p1688552733333679?thread_ts=1688542528.385169&cid=C015VQQ3HMG)

Solution: create index concurrently

Deployment notes:
- [x] Test on local
- [x] Test on staging

### CODE REVIEW

File: backend/src/database/migrations/20230705102914-index-sms-transactional-message-id-column.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = {
+up: async (queryInterface, _) => {
```
Comment (blakegong) [inline backend/src/database/migrations/20230705102914-index-sms-transactional-message-id-column.js:4]: /nit alignment (you guys have no linting huh 😂)

**Review by blakegong — APPROVED** [review]

File: backend/src/database/migrations/20230705102914-index-sms-transactional-message-id-column.js (hunk 1 above)
Comment (tan-zx) [reply backend/src/database/migrations/20230705102914-index-sms-transactional-message-id-column.js:4]: lol weird, we do, but somehow it's not being applied. thanks, will fix.

**Comment by tan-zx** [comment]
Previously have tried to add `@Index`, but it doesn't play well with how we set up our test database using `sequelize.sync()`:
- [commit adding `@Index`](https://github.com/opengovsg/postmangovsg/pull/2070/commits/672b6b79e454b7afbcb866843c55a6acc72dca81)
- [commit removing it](https://github.com/opengovsg/postmangovsg/pull/2070/commits/3698734cf37d9ea72bcbf3490e95ee5c83ed91fd)
- error countered:
image

File: backend/src/sms/models/sms-message-transactional.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ export class SmsMessageTransactional extends Model<SmsMessageTransactional> {
   @Column({ type: DataType.TEXT, allowNull: false })
   body: string
 
+  // indexed for faster search
   @Column({ type: DataType.STRING, allowNull: true })
```
Comment (stanleynguyen) [inline backend/src/sms/models/sms-message-transactional.ts:48]: Can add `unique: true` here instead of commenting actually

Comment (tan-zx) [reply backend/src/sms/models/sms-message-transactional.ts:48]: by right should add `@Index` decorator, but [couldn't get it to work](https://github.com/opengovsg/postmangovsg/pull/2077#issuecomment-1621530115). same issue encountered when I was adding an index for WhatsApp's `message_id` column.

**Pull request merged**